### PR TITLE
Identity fanout

### PR DESCRIPTION
**Pull request by manfre**
Identity update/delete fanout.

### CODE REVIEW

File: users/admin.py [hunk 1]
```diff
@@ -75,8 +75,8 @@ class FollowAdmin(admin.ModelAdmin):
     list_filter = [LocalSourceFilter, LocalTargetFilter, "state"]
     raw_id_fields = ["source", "target"]
 
-    def has_add_permission(self, request, obj=None):
-        return False
+    # def has_add_permission(self, request, obj=None):
+    #     return False
```
Comment (manfre) [inline users/admin.py:79]: This was done for ad-hoc testing. Will clean up.

File: users/models/identity.py [hunk 2]
```diff
@@ -32,19 +32,71 @@
...
+
+    When a local identity is "edited" or "deleted", it will fanout the change to
+    all followers and transition to "updated"
     """
 
     outdated = State(try_interval=3600, force_initial=True)
     updated = State(try_interval=86400 * 7, attempt_immediately=False)
 
+    edited = State(try_interval=300, attempt_immediately=True)
+    # deleted = State(externally_progressed=True)
```
Comment (manfre) [inline users/models/identity.py:46]: Not sure if we want a separate delete or let it flow through as an edit that reacts to `Identity.deleted`

Comment (andrewgodwin) [reply users/models/identity.py:46]: Separate delete, as that way we can also use the state as a "is this user deleted" filter

**Review by andrewgodwin — COMMENTED** [review]
Looks good apart from the two areas you highlighted

File: users/models/identity.py [hunk 3]
```diff
@@ -30,19 +30,71 @@
...
+
+            await FanOut.objects.acreate(
+                identity=follower.source,
+                type=type_,
+                subject_identity=identity,
+            )
+            shared_inboxes.add(shared_uri)
+
+    @classmethod
+    async def handle_new(cls, instance):
```
Comment (andrewgodwin) [inline users/models/identity.py:76]: There's no "new" state here, so why is this handler here?

File: users/models/identity.py [hunk 4]
```diff
@@ -30,19 +30,71 @@
...
+    @classmethod
+    async def handle_edited(cls, instance: "Identity"):
+        from activities.models import FanOut
+
+        if not instance.local:
+            return cls.updated
+
+        identity = await instance.afetch_full()
+
+        if identity.deleted:
```
Comment (andrewgodwin) [inline users/models/identity.py:88]: While I did leave a "deleted" field on the model, I'd rather this follows the existing Post pattern where we actually use "deleted" and "deleted_fanned_out" states, and we ignore the "deleted" field on the model entirely (and remove it when the next migration opportunity arises)

**Review by andrewgodwin — COMMENTED** [review]
Couple of small requests. Also worth noting - Mastodon fans out profile deletes to every single server it knows about (presumably in case they are cached locally), but I don't know if we need to do the same yet.

File: users/models/identity.py (hunk 3 above)
Comment (manfre) [reply users/models/identity.py:76]: Remnants after a rebase.

File: users/models/identity.py (hunk 4 above)
Comment (manfre) [reply users/models/identity.py]: Changed flow to `deleted` and `deleted_fanned_out`.

**Review by andrewgodwin — APPROVED** [review]

**Pull request merged**